rust/xaynet-analytics/src/database/common.rs

```diff
@@ -1,35 +1,55 @@
 use anyhow::{anyhow, Error, Result};
 use isar_core::{
     index::StringIndexType,
-    object::{data_type::DataType, object_builder::ObjectBuilder},
+    object::{
+        data_type::DataType,
+        isar_object::{IsarObject, Property},
+        object_builder::ObjectBuilder,
+    },
     schema::collection_schema::CollectionSchema,
 };
-use std::vec::IntoIter;
+use std::{convert::TryFrom, vec::IntoIter};
 
 use crate::database::isar::IsarDb;
 
-pub trait IsarAdapter: Sized {
+pub trait IsarAdapter<'object>: Sized {
     fn into_field_properties() -> IntoIter<FieldProperty>;
```

Comment:
definitely yes, this [task/story](https://xainag.atlassian.net/browse/XN-1517) should be about that: removing the need for a lot of the `Vec`s there are currently there and turn them into iterators, and it could be generalised to these field properties too, not only in `DataCombiner`. Again, i don't think it'll be too difficult, just need some time